File: src/entity/Condition.ts
/**
 * @main Condition
 */

import { Errata } from "./_Erratum"
import { PublicationRefs } from "./_PublicationRef"

/**
 * @title Condition
 */
export type Condition = {
  /**
   * An unique, increasing integer.
   * @integer
   * @minimum 1
   */
  id: number

  src: PublicationRefs

  /**
   * All translations for the entry, identified by IETF language tag (BCP47).
   * @minProperties 1
   */
  translations: {
    /**
     * @patternProperties ^[a-z]{2}-[A-Z]{2}$
     */
    [localeId: string]: Translation
  }
}

type Translation = {
  /**
   * The name of the condition.
   * @minLength 1
   */
  name: string

  /**
   * Additional rules for the condition, if applicable.
   * @markdown
   * @minLength 1
   */
  rules?: string

  /**
   * The effects for level 1 to 4.
   */
  effects: Effects

  errata?: Errata
}

/**
 * The effects on levels 1 to 4.
 */
type Effects = [Effect, Effect, Effect, Effect]

/**
 * The effect on a level.
 * @markdown
 * @minLength 1
 */
type Effect = string

File: Sources/OptolithDatabaseSchema/GeneratedTypes/Condition.swift
//
//  Condition.swift
//  OptolithDatabaseSchema
//

import DiscriminatedEnum

public struct Condition: LocalizableEntity {
    /// The condition's identifier. An unique, increasing integer.
    public let id: Int
    
    public let src: PublicationRefs
    
    /// All translations for the entry, identified by IETF language tag (BCP47).
    public let translations: LocaleMap<ConditionTranslation>

    public init(id: Int, src: PublicationRefs, translations: LocaleMap<ConditionTranslation>) {
        self.id = id
        self.src = src
        self.translations = translations
    }
}

public struct ConditionTranslation: EntitySubtype {
    /// The condition's name.
    public let name: NonEmptyString
    
    /// Additional rules for the condition, if applicable.
    public let rules: NonEmptyMarkdown?
    
    /// The effects for level 1 to 4.
    public let effects: [NonEmptyMarkdown]
    
    public let errata: Errata?

    public init(name: NonEmptyString, rules: NonEmptyMarkdown? = nil, effects: [NonEmptyMarkdown], errata: Errata? = nil) {
        self.name = name
        self.rules = rules
        self.effects = effects
        self.errata = errata
    }
}
